tozny/provider.go

```diff
@@ -2,14 +2,103 @@
 package tozny
 
 import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	toznyClients "github.com/tozny/e3db-clients-go"
+	accountClient "github.com/tozny/e3db-clients-go/accountClient"
 )
 
-// Provider returns a Terraform Provider for automating provisioning of Tozny Services using Terraform.
+// Provider returns the Terraform schema for the Tozny provider for automating provisioning of Tozny Services using Terraform.
 func Provider() *schema.Provider {
 	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"api_endpoint": &schema.Schema{
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://api.e3db.com",
+			},
+			"username": &schema.Schema{
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("TOZNY_ACCOUNT_USERNAME", nil),
+			},
+			"password": &schema.Schema{
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
Should this be required?

Reply:
no because there are things you can do with just client (not account) level credentials, and even things (like create an account) that can be done with no credentials